code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -109,6 +123,46 @@
 	. += emissive_appearance(icon, "locked", alpha = src.alpha)
 	. += locked ? "locked" : "unlocked"
 
+/// Animates the closet door opening and closing
+/obj/structure/closet/proc/animate_door(closing = FALSE)
+	if(door_anim_time)
+		if(!door_obj)
+			door_obj = new
+		vis_contents += door_obj
+		door_obj.icon = icon
+		door_obj.icon_state = "[icon_door || icon_state]_door"
+		is_animating_door = TRUE
+		var/num_steps = door_anim_time / world.tick_lag
+		for(var/step in 0 to num_steps)
+			var/angle = door_anim_angle * (closing ? 1 - (step/num_steps) : (step/num_steps))
+			var/matrix/door_transform = get_door_transform(angle)
+			var/door_state = angle >= 90 ? "[icon_door_override ? icon_door : icon_state]_back" : "[icon_door || icon_state]_door"
```

Comment:
and another here? could we just like, store the base icon name to use instead of doing these checks constantly?

Reply:
I can only find icon_door_override in closet.dm, I suspect that it's never set to true 🤔